### PR TITLE
Matmul overloaded for correlator class.

### PR DESCRIPTION
**Pull request by fjosw**
I overloaded the matmul class which should now allow for the multiplication of matrix valued `Corr` objects with matrices. I will write a few tests and look into edge cases but this solution seems to work.

### CODE REVIEW

**Comment by fjosw** [comment]
I know extended the `matmul` method to also work on two correlator objects.

Additionally, I added a `trace` method for matrix correlators.

**Comment by fjosw** [comment]
Another bug in `Corr.roll` for matrix correlators fixed.

File: pyerrors/correlators.py [hunk 1]
```diff
@@ -254,6 +254,18 @@ def is_matrix_symmetric(self):
                         return False
         return True
 
+    def trace(self):
+        """Calculates the per-timeslice trace of a correlator matrix."""
+        if self.N == 1:
+            raise TypeError("Only works for correlator matrices.")
```
Comment (s-kuberski) [inline pyerrors/correlators.py:260]: Strictly speaking, it is not necessary to catch this, since the trace of a 1x1 matrix should be well defined. But of course it does not make much sense.

File: pyerrors/correlators.py [hunk 2]
```diff
@@ -405,7 +417,7 @@ def roll(self, dt):
         dt : int
             number of timeslices
         """
-        return Corr(list(np.roll(np.array(self.content, dtype=object), dt)))
+        return Corr(list(np.roll(np.array(self.content, dtype=object), dt, axis=0)))
```
Comment (s-kuberski) [inline pyerrors/correlators.py:420]: Very nice!

**Comment by s-kuberski** [comment]
Hi,

I did not (yet?) check the code by running it, but so-far everything looks fine. Thanks for taking care of this.

While you are at it: Currently, it is not possible to initialize a `Corr` object from a one-dimensional array:
```python
li = [pe.cov_Obs(o, 0, '') for o in [1, 2, 3, 4]]
ar = np.array(li)
pe.Corr(li)
pe.Corr(ar)
```
```
----> 4 pe.Corr(ar)

     46             # This only works, if the array fulfills the conditions below
---> 47             if not len(data_input.shape) == 2 and data_input.shape[0] == data_input.shape[1]:
     48                 raise Exception("Incompatible array shape")
     49             if not all([isinstance(item, Corr) for item in data_input.flatten()]):

IndexError: tuple index out of range
```
would you mind looking into this?

File: pyerrors/correlators.py (hunk 1 above)
Comment (fjosw) [reply pyerrors/correlators.py:260]: I just checked and numpy does not like the trace of an array with a single entry:
```python
np.trace(np.array([1.1]))
> ValueError: diag requires an array of at least two dimensions
```
